chisel/src/executor.rs

```diff
@@ -0,0 +1,766 @@
+                Paint::cyan(format!("0x{:x}", i)),
+                Paint::cyan(I256::from_raw(i))
+            )
+        }
+        Token::Uint(i) => {
+            format!(
+                "Type: {}\n├ Hex: {}\n└ Decimal: {}",
+                Paint::red("uint"),
+                Paint::cyan(format!("0x{:x}", i)),
+                Paint::cyan(i)
+            )
+        }
+        Token::Bool(b) => {
+            format!("Type: {}\n└ Value: {}", Paint::red("bool"), Paint::cyan(b))
+        }
+        Token::String(_) | Token::Bytes(_) => {
+            let hex = hex::encode(ethers::abi::encode(&[token.clone()]));
+            let s = token.into_string();
+            format!(
+                "Type: {}\n{}├ Hex (Memory):\n├─ Length ({}): {}\n├─ Contents ({}): {}\n├ Hex (Tuple Encoded):\n├─ Pointer ({}): {}\n├─ Length ({}): {}\n└─ Contents ({}): {}",
+                Paint::red(if s.is_some() { "string" } else { "dynamic bytes" }),
+                if s.is_some() {
+                    format!("├ UTF-8: {}\n", Paint::cyan(s.unwrap()))
+                } else {
+                    String::default()
+                },
+                Paint::yellow("[0x00:0x20]"),
+                Paint::cyan(format!("0x{}", &hex[64..128])),
+                Paint::yellow("[0x20:..]"),
+                Paint::cyan(format!("0x{}", &hex[128..])),
+                Paint::yellow("[0x00:0x20]"),
+                Paint::cyan(format!("0x{}", &hex[..64])),
+                Paint::yellow("[0x20:0x40]"),
+                Paint::cyan(format!("0x{}", &hex[64..128])),
+                Paint::yellow("[0x40:..]"),
+                Paint::cyan(format!("0x{}", &hex[128..])),
+            )
+        }
+        Token::FixedArray(tokens) | Token::Array(tokens) => {
+            let mut out = format!(
+                "{}({}) = {}",
+                Paint::red("array"),
+                Paint::yellow(format!("{}", tokens.len())),
+                Paint::red('[')
+            );
+            for token in tokens {
+                out.push_str("\n  ├ ");
+                out.push_str(&format_token(token).replace('\n', "\n  "));
+                out.push('\n');
+            }
+            out.push_str(&Paint::red(']').to_string());
+            out
+        }
+        Token::Tuple(tokens) => {
+            let mut out = format!(
+                "{}({}) = {}",
+                Paint::red("tuple"),
+                Paint::yellow(tokens.iter().map(|t| t.to_string()).collect::<Vec<_>>().join(",")),
+                Paint::red('(')
+            );
+            for token in tokens {
+                out.push_str("\n  ├ ");
+                out.push_str(&format_token(token).replace('\n', "\n  "));
+                out.push('\n');
+            }
+            out.push_str(&Paint::red(')').to_string());
+            out
+        }
+    }
+}
+
+// Ripped from
+// [soli](https://github.com/jpopesculian/soli)
+// =============================================
+
+#[derive(Debug, Clone)]
```

Comment:
suggestion: move to standalone file

Reply:
Can do- this logic is only used within `executor.rs`. LMK if you'd still like this pulled out into its own file 😄